source/distributed/tria.cc

```diff
@@ -4285,10 +4286,10 @@ namespace parallel
     template <int dim, int spacedim>
     TemporarilyMatchRefineFlags<dim, spacedim>::TemporarilyMatchRefineFlags(
       dealii::Triangulation<dim, spacedim> &tria)
-      : distributed_tria(
-          dynamic_cast<
-            dealii::parallel::distributed::Triangulation<dim, spacedim> *>(
-            &tria))
+      :
+      [[maybe_unused]] distributed_tria(
```

Comment:
You cannot add the attribute to a class member initialization in the constructor.

Reply:
The attribute has been removed from class member initialization in the constructor.